### DISCUSSION TITLE
Can I directly configure a word dictionary for meilisearch?

### DISCUSSION THREAD

**Discussion by jet10000** [meilisearch/product#135] [Feedback & Feature Proposal]
Can I directly configure a word dictionary for meilisearch? 

Do not use word segmentation technology, directly use the word list, for cjk.

**Comment by dichotommy** [comment]
Hi @jet10000 , sorry for the delay in answering your question. For now, the answer is no. MeiliSearch uses a tokenizer, so there's no easy way to substitute a dictionary.

However, the tokenizer has been redone for the latest release and should offer much improved performance w/ CJK thanks to Jieba. Give it a try and let us know what you think!

**Comment by jet10000** [comment]
@react-learner 
Is there a `match_phrase` query function of elasticsearch, because the jieba word segmentation also divides the words that should not be divided, which is very subjective. Or is it possible to restrict and configure jieba word segmentation.

**Comment by jet10000** [comment]
If you want to use meilisearch in a special field or industry, the industry vocabulary dictionary should allow configuration.

**Comment by jet10000** [comment]
Highlight, whether the searched keywords can be highlighted in red, but the keywords after the word segmentation are displayed in green, or just only highlight search keywords that not be segmented.

red: `咳嗽`
green: `咳`  and `嗽`

**Comment by dichotommy** [comment]
@jet10000 
> Is there a `match_phrase` query function of elasticsearch, because the jieba word segmentation also divides the words that should not be divided, which is very subjective. Or is it possible to restrict and configure jieba word segmentation.

For the moment, there is no `match_phrase` equivalent. You might be able to achieve some similar results by reordering the [ranking rules](https://docs.meilisearch.com/guides/main_concepts/relevancy.html#ranking-rules) and putting `exactness` and `proximity` first, but it's not really the same since you will still see out of order results.

> If you want to use meilisearch in a special field or industry, the industry vocabulary dictionary should allow configuration.

To my knowledge, MeiliSearch is not really intended for this use. We follow a zero-config design principle, so even though we want to make the software as flexible/customizable as possible, out-of-the-box functionality will always come first.

Who knows though, more precise tokenizer customization could be a feature in the future. The new tokenizer is modular so that already lays the groundwork for users to develop and add their own tokenization pipelines.

> Highlight, whether the searched keywords can be highlighted in red, but the keywords after the word segmentation are displayed in green, or just only highlight search keywords that not be segmented.

This is a pretty good idea to differentiate exact, ordered matches from partial or out-of-order matches (although as a colorblind person, please don't use red and green).

**Comment by gmourier** [comment]
Closing this discussion in favor of #663
